ownCloudSDK/Logging/Writers/OCLogFileWriter.m

```diff
@@ -144,6 +267,78 @@ - (nullable NSError *)eraseOrTruncate
 	return (error);
 }
 
+- (NSDictionary*)_attributesForPath:(NSString*)path
+{
+	NSDictionary *attributes = nil;
+
+	if ([[NSFileManager defaultManager] fileExistsAtPath:path])
+	{
+		attributes = [[NSFileManager defaultManager] attributesOfItemAtPath:path error:nil];
+	}
+
+	return attributes;
+}
+
+- (void)_scheduleLogRotationTimer
+{
+	[self _unscheduleLogRotationTimer];
+	_logRotationTimerSource = dispatch_source_create(DISPATCH_SOURCE_TYPE_TIMER, 0, 0, [OCLogWriter queue]);
+
+	__weak __auto_type weakSelf = self;
+	dispatch_source_set_event_handler(_logRotationTimerSource, ^{
+		[weakSelf _rotateLogIfRequired];
+	});
+
+	dispatch_time_t fireTime = dispatch_time(DISPATCH_TIME_NOW, OCDefaultLogRotationFrequency);
+
+	dispatch_source_set_timer(_logRotationTimerSource, fireTime, DISPATCH_TIME_FOREVER, 1ull * NSEC_PER_SEC);
+	dispatch_resume(_logRotationTimerSource);
+}
+
+- (void)_unscheduleLogRotationTimer
+{
+	if (_logRotationTimerSource) {
+		dispatch_source_cancel(_logRotationTimerSource);
+		_logRotationTimerSource = NULL;
+	}
+}
+
+- (void)_rotateLogIfRequired
+{
+	NSError *error = nil;
+	NSDate *logCreationDate = [self _attributesForPath:self.logFileURL.path][NSFileCreationDate];
+
+	if (logCreationDate != nil)
+	{
+		// Check the age of current log
+		NSTimeInterval age = [logCreationDate timeIntervalSinceNow];
+		if (-age > self.rotationInterval)
+		{
+			// Construct path for the archived log
+			NSString *timeStamp = [OCLogWriter timestampStringFrom:[NSDate date]];
+			NSString *transformedTimestamp = [timeStamp stringByReplacingOccurrencesOfString:@"[ /:]+" withString:@"_" options:NSRegularExpressionSearch range:NSMakeRange(0, [timeStamp length])];
+
+			NSString *arhivedLogPath = [self.logFileURL.path stringByAppendingFormat:@".%@", transformedTimestamp];
```

Comment:
Little typo here: should be ar`c`hivedLogPath.

Reply:
@felix-schwarz fixed